src/jgikbase/idmapping/util/util.py

```diff
@@ -17,8 +19,39 @@ def not_none(obj: object, name: str):
     :param obj: the object to check
     :param name: the name of the object to use in error messages.
     """
-    if type(obj) == str:
-        if not cast(str, obj).strip():
-            raise ValueError(name + ' cannot be None or whitespace only')
-    elif not obj:
+    if not obj:
         raise ValueError(name + ' cannot be None')
+
+# TODO EXCEP change exceptions to package specific
+
+
+_REGEX_CACHE: _Dict[str, _Pattern] = {}
```

Comment:
I'd consider removing this, as I would bet that compiling small regexes without a cache will have zero impact on the performance of your app

Reply:
Probably. It also doesn't hurt, so *shrug*